Seasons Defense/Assets/Scripts/TowerController.cs

```diff
@@ -4,24 +4,21 @@ public class TowerController : MonoBehaviour
 {
     public GameObject missilePrefab;
     public float offset;
+    
+    public void Fire() {
 
-    private void Update()
-    {
-        if (Input.GetMouseButtonDown(1))
-        {
-            Vector3 mousePosition = new Vector3(Input.mousePosition.x, Input.mousePosition.y, Camera.main.nearClipPlane);
-            Vector3 worldPosition = Camera.main.ScreenToWorldPoint(mousePosition);
-            worldPosition.z = transform.position.z;
-            GameObject missileInstance = Instantiate(missilePrefab, new Vector3(transform.position.x, transform.position.y + offset, 0), Quaternion.identity);
-
-            // Give the missile the attributes it needs to move towarsd its target position
-            Missile missile = missileInstance.GetComponent<Missile>();
+        Vector3 mousePosition = new Vector3(Input.mousePosition.x, Input.mousePosition.y, Camera.main.nearClipPlane);
+        Vector3 worldPosition = Camera.main.ScreenToWorldPoint(mousePosition);
+        worldPosition.z = transform.position.z;
+        GameObject missileInstance = Instantiate(missilePrefab, new Vector3(transform.position.x, transform.position.y + offset, 0), Quaternion.identity);
+      
+        // Give the missile the attributes it needs to move towarsd its target position
+        Missile missile = missileInstance.GetComponent<Missile>();
 
-            if (missile != null)
-            {
-                missile.speed = 5;
-                missile.target = worldPosition;
-            }
+        if (missile != null)
+        {
+            missile.speed = 5;
```

Comment:
This speed is set after instantiation of the object which may call Update() on itself without a set speed value before this conditional statement. Consider initializing this in Missile.cs, but you could keep this set here as well.